flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterDescriptor.java

```diff
@@ -292,26 +292,39 @@ public void addShipFiles(List<File> shipFiles) {
         this.shipFiles.addAll(shipFiles);
     }
 
-    private void addShipArchives(List<File> shipArchives) {
+    private void addShipArchives(List<Path> shipArchives) {
         checkArgument(
-                isArchiveOnlyIncludedInShipArchiveFiles(shipArchives),
+                isArchiveOnlyIncludedInShipArchiveFiles(shipArchives, yarnConfiguration),
                 "Non-archive files are included.");
         this.shipArchives.addAll(shipArchives);
     }
 
-    private static boolean isArchiveOnlyIncludedInShipArchiveFiles(List<File> shipFiles) {
-        return shipFiles.stream()
-                .filter(File::isFile)
-                .map(File::getName)
-                .map(String::toLowerCase)
-                .allMatch(
-                        name ->
-                                name.endsWith(".tar.gz")
-                                        || name.endsWith(".tar")
-                                        || name.endsWith(".tgz")
-                                        || name.endsWith(".dst")
-                                        || name.endsWith(".jar")
-                                        || name.endsWith(".zip"));
+    private static boolean isArchiveOnlyIncludedInShipArchiveFiles(List<Path> shipFiles, YarnConfiguration yarnConfig) {
+        for (Path shipFile : shipFiles) {
+            try {
+                if (Utils.isRemotePath(shipFile.toString())) {
+                    final FileSystem fs = shipFile.getFileSystem(yarnConfig);
+                    final String name = shipFile.getName().toLowerCase();
+                    if (fs.isFile(shipFile) && !(name.endsWith(".tar.gz") || name.endsWith(".tar") || name.endsWith(
+                        ".tgz") || name.endsWith(".dst") || name.endsWith(".jar") || name.endsWith(".zip"))) {
+                        return false;
+                    }
+                } else {
+                    final File localFile = new File(shipFile.toUri().getPath());
+                    final String name = localFile.getName().toLowerCase();
+                    if (localFile.isFile() && !(name.endsWith(".tar.gz") || name.endsWith(".tar") || name.endsWith(
+                        ".tgz") || name.endsWith(".dst") || name.endsWith(".jar") || name.endsWith(".zip"))) {
+                        return false;
+                    }
+                }
+            } catch (IOException e) {
+                LOG.error("The shipping archive archive {} parsing failed.", shipFile);
+                throw new RuntimeException("The shipping archive " + shipFile + " parsing failed. " +
+                    "Error message: " + e.getMessage());
```

Comment:
We should not throw `RuntimeException` but rather fail explicitly if an error occurs here. We need to tell the user that something is wrong with his shipped files.